src/js/providers/tracks-mixin.js

```diff
@@ -551,10 +551,7 @@ define(['utils/underscore',
             // already have one with the same label
             track = _.findWhere(tracks, { label: label });
 
-            if (track) {
-                track.kind = itemTrack.kind;
-                track.language = itemTrack.language || '';
-            } else {
```

Comment:
Do we depend on `kind` being updated properly in other places?

You should search the code base for places where we read `kind`. If necessary we should consider adding our own property using `Object.defineProperty`.